### PR TITLE
Token Auth v1

### PR DESCRIPTION
**Pull request by magnetised**
just enough auth to require a real token in the authentication request, giving the ws server access to the user id

Also closes VAX-303

### CODE REVIEW

**Comment by define-null** [comment]
https://github.com/electric-sql/electric/pull/55 is merged, please rebase to rerun e2e tests

**Comment by define-null** [comment]
We do not seem to use mockserver anymore, lets remove it then.

**Comment by magnetised** [comment]
@define-null see https://github.com/electric-sql/console/pull/7 for the console side of things, including new env var for the database_id stuff

File: lib/electric/satellite/satellite_protocol.ex [hunk 1]
```diff
@@ -128,20 +132,22 @@ defmodule Electric.Satellite.Protocol do
...
+          {:ok, auth} ->
+            reg_name = Electric.Satellite.WsServer.reg_name(state.client)
+            Electric.reg(reg_name)
+            :ok = ClientManager.register_client(state.client, reg_name)
 
-        with :ok <- Electric.Satellite.Auth.validate_token(id, token) do
-          reg_name = Electric.Satellite.WsServer.reg_name(state.client)
-          Electric.reg(reg_name)
-          :ok = ClientManager.register_client(state.client, reg_name)
+            {%SatAuthResp{id: "server_identity"},
```
[the remark this answers is not in the document]
Comment (icehaunter) [reply lib/electric/satellite/satellite_protocol.ex:148]: @magnetised, `database_id` is much better off being named `global_cluster_id` everywhere, now that we've merged the ID distinctions into console.

File: lib/electric/satellite/satellite_protocol.ex [hunk 2]
```diff
@@ -128,20 +132,22 @@ defmodule Electric.Satellite.Protocol do
...
+    %{database_id: database_id} = state
+
     case msg do
-      %SatAuthReq{id: id, token: token}
-      when id !== "" and token !== "" ->
-        Logger.debug("Received auth request #{inspect(state.client)} for #{inspect(id)}")
+      %SatAuthReq{id: client_id, token: token} when client_id !== "" and token !== "" ->
+        Logger.debug("Received auth request #{inspect(state.client)} for #{inspect(client_id)}")
+
+        case Electric.Satellite.Auth.validate_token(database_id, token) do
```
[the remark this answers is not in the document]
Comment (magnetised) [reply lib/electric/satellite/satellite_protocol.ex:142]: good point. it's not like it's hard to get the cluster id at any point.

File: lib/electric/satellite/satellite_auth.ex [hunk 3]
```diff
@@ -2,62 +2,110 @@ defmodule Electric.Satellite.Auth do
...
+  @spec validate_token(String.t(), String.t()) ::
+          {:ok, t()} | {:error, :auth_not_configured | term()}
+  def validate_token(database_id, token) do
+    case validate_jwt_token(database_id, token) do
+      {:ok, auth} ->
+        {:ok, auth}
 
       {:error, reason} = error ->
-        Logger.error("authorization failed for #{id} with reason: #{inspect(reason)}")
+        Logger.error("authorization failed for #{database_id} with reason: #{inspect(reason)}")
```
[the remark this answers is not in the document]
Comment (magnetised) [reply lib/electric/satellite/satellite_auth.ex]: have lowered the level for expired tokens, there was a way...

I've also updated the logger metadata to include client id and user id

**Review by define-null — CHANGES_REQUESTED** [review]
Looks good to me! There are some tests still failing both on ExUnit and E2E side.

**Comment by define-null** [comment]
Failure on ws is caused by LSN recent change and it could be ignored.

**Review by define-null — APPROVED** [review]

**Pull request merged**